src/controllers/v1/entities.js

```diff
@@ -57,28 +58,279 @@ module.exports = class Entities extends Abstract {
   ]
 */
 
-  /**
-   * List all the entities.
-   * @method
-   * @name list 
-   * @returns {JSON} - List of all entities.
-   */
+	/**
+	 * Find all the entities.
+	 * @method
+	 * @name find
+	 * @returns {JSON} - List of all entities.
+	 */
+
+	find(req) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let entityData = await entitiesHelper.entityDocuments(req.body.query, req.body.projection)
+
+				return resolve({
+					message: CONSTANTS.apiResponses.ENTITY_FETCHED,
+					result: entityData,
+				})
+			} catch (error) {
+				return reject({
+					status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+					message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+					errorObject: error,
+				})
+			}
+		})
+	}
+
+
+	details(req) {
+		console.log(req.query.type,"line no 89");
+		console.log(req.params._id,"line no 90");
+		return new Promise(async (resolve, reject) => {
+	
+		  try {
+	
+			let result = await entitiesHelper.details(
+			  req.query.type, 
+			  req.params._id
+			);
+	
+			return resolve({
+			  message: CONSTANTS.apiResponses.ENTITY_INFORMATION_FETCHED,
+			  result: result
+			});
+	
+		  } catch (error) {
+	
+			return reject({
+			  status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+			  message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+			  errorObject: error
+			})
+	
+		  }
+	
+	
+		})
+	  }
+	
+
+	async create(req) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let result = await entitiesHelper.create(req.body, req.userDetails)
+				return resolve({
+					message: CONSTANTS.apiResponses.ENTITY_ADDED,
+					result: result,
+				})
+			} catch (error) {
+				console.log(error,"line no 130");
+				return reject({
+					status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+					message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+					errorObject: error,
+				})
+			}
+		})
+	}
+
+
+	update(req) {
+		return new Promise(async (resolve, reject) => {
+		  try {
+			let result = await entitiesHelper.update( req.params._id, req.body);
+	
+			return resolve({
+			  message: CONSTANTS.apiResponses.ENTITY_INFORMATION_UPDATE,
+			  result: result
+			});
+	
+		  } catch (error) {
+	
+			return reject({
+			  status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+			  message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+			  errorObject: error
+			})
+	
+		  }
+	
+	
+		})
+	  }
+	listByEntityType(req) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let schemaMetaInformation = entitiesHelper.entitiesSchemaData().SCHEMA_METAINFORMATION
+				let projection = [
+					schemaMetaInformation + '.externalId',
+					schemaMetaInformation + '.name',
+					'registryDetails.locationId',
+				]
+
+
+				let skippingValue = req.pageSize * (req.pageNo - 1)
+				let entityDocuments = await entitiesHelper.entityDocuments(
```

Comment:
@MallanagoudaB create a folder databaseQueries and entities file inside that. Then add all the db queries there